src/data-access/repositories/mongo/mongo-ingested-part-repository.ts

```diff
@@ -0,0 +1,57 @@
+import { BaseMongoRepository } from './base-mongo-repository'
+import { IngestedPartRepository } from '../interfaces/ingested-part-repository'
+import { IngestedPart } from '../../../model/entities/ingested-part'
+import { MongoDatabase } from './mongo-database'
+import { MongoIngestedEntityConverter, MongoIngestedPart } from './mongo-ingested-entity-converter'
+import { IngestedPieceRepository } from '../interfaces/ingested-piece-repository'
+import { NotFoundException } from '../../../model/exceptions/not-found-exception'
+
+const INGESTED_PART_COLLECTION_NAME: string = 'parts' // TODO: Once we control ingest rename to "ingestedParts"
+
+export class MongoIngestedPartRepository extends BaseMongoRepository implements IngestedPartRepository {
+
+  constructor(
+    mongoDatabase: MongoDatabase,
+    private readonly mongoIngestedEntityConverter: MongoIngestedEntityConverter,
+    private readonly ingestedPieceRepository: IngestedPieceRepository
+  ) {
+    super(mongoDatabase)
+  }
+
+  protected getCollectionName(): string {
+    return INGESTED_PART_COLLECTION_NAME
+  }
+
+  public async getIngestedPart(partId: string): Promise<IngestedPart> {
+    this.assertDatabaseConnection(this.getIngestedPart.name)
+    const mongoIngestedPart: MongoIngestedPart | null = await this.getCollection().findOne<MongoIngestedPart>({
+      _id: partId
+    })
+    if (!mongoIngestedPart) {
+      throw new NotFoundException(`No Part found for partId: ${partId}`)
+    }
+    const ingestedPart: IngestedPart = this.mongoIngestedEntityConverter.convertIngestedPart(mongoIngestedPart)
+    ingestedPart.ingestedPieces = await this.ingestedPieceRepository.getIngestedPieces(ingestedPart.id)
+    return ingestedPart
+  }
+
+  public async getIngestedParts(segmentId: string): Promise<IngestedPart[]> {
```

Comment:
This comes down to your naming philosofy. 
One philosofy is to have everything in the name. Another philosofy is to have the parameter as part of the "name".
So by reading the parameter in conjunction with the method name, you would get that this method "get ingested parts for a segment".

As of the current implementation, the id of the IngestedSegment is the same as the id of Segment. Furthermore, the attribute is currently called `IngestedPart.segmentId`. Because of these reasons, I am going to leave it as is for now.

Reply:
It's been updated